tests/posix/env/prj.conf

```diff
@@ -1,3 +1,4 @@
 CONFIG_ZTEST=y
 CONFIG_POSIX_ENV=y
+CONFIG_PICOLIBC=y
```

Comment:
Given that Picolibc should be the default now on most platforms that do not require a specific C library, I think my preference would be to add this to `boards/native_sim.conf` and `boards/native_sim_64.conf`.

Reply:
There is more native boards than those 2. So we'd end up with lots of overlays. Setting it here does not change anything for any non native target as it is already their default.
At least I don't see any technical drawback to do it here. So I'd ask you reconsider.

Reply:
I wanted to double check about random configurations like arc or llvm or when picolibc is not available as a module, but it looks fine.